Sources/SwiftFormat/PrettyPrint/TokenStreamCreator.swift

```diff
@@ -3556,7 +3556,12 @@ fileprivate final class TokenStreamCreator: SyntaxVisitor {
 
       case .spaces(let n):
         guard leadingIndent == .spaces(0) else { break }
-        leadingIndent = .spaces(n)
+        if config.indentBlankLines, trivia.count > index + 1, trivia[index + 1].isNewline {
+          appendToken(.space(size: n))
+          requiresNextNewline = true
```

Comment:
Does this mean that superfluous newline are no longer discarded if `config.indentBlankLines` is true? That seems surprising to me. Eg. 

```swift
func foo() {}


func bar() {}
```

should still be formatted to only have one newline.

Reply:
Looking at it again, this logic doesn’t seem appropriate 🤔 
In the initial implementation, this was added along with logic to retain user-entered spaces by inserting a `space` token, but it doesn’t seem necessary with the new implementation approach. 
I’ll keep the logic that adds spaces as `leadingIndent` and, when processing newlines, add a soft newline if there’s a `leadingIndent`, ensuring that the actual output doesn’t exceed the configured limit lines.